python/ola/RDMTest.py

```diff
@@ -101,6 +102,61 @@ def ResponseCallback(self, response, data, unpack_exception):
     self.assertTrue(results.gotrequest)
     self.assertTrue(results.gotresponse)
 
+  # @timeout_decorator.timeout(2)
+  def testGetParamsWithResponse(self):
+    """uses client to send an RDM get with mocked olad.
+    Regression test that confirms sent message is correct and
+    sends fixed response message."""
+    sockets = socket.socketpair()
+    wrapper = ClientWrapper(sockets[0])
+    pid_store = PidStore.GetStore(pidStorePath)
+    client = wrapper.Client()
+    rdm_api = RDMAPI(client, pid_store)
+
+    class results:
+      gotrequest = False
+      gotresponse = False
+
+    def DataCallback(self):
+      # request and response for
+      # ola_rdm_get.py -u 1 --uid 7a70:ffffff00 parameter_description 17
+      # against olad dummy plugin
+      # enable logging in rpc/StreamRpcChannel.py
+      data = sockets[1].recv(4096)
+      expected = binascii.unhexlify(
+        "2b000010080110001a0a52444d436f6d6d616e6422190801120908f0f4011500"
+        "ffffff180020512a02001130003800")
+      self.assertEqual(data, expected,
+                       msg="Regression check failed. If protocol change "
+                       "was intended set expected to: " +
+                       str(binascii.hexlify(data)))
+      results.gotrequest = True
+      response = binascii.unhexlify(
+        "2e000010080210002228080010021800220200062851300038004a0908f0f401"
+        "1500ffffff520908f0f40115ac107de05811")
+      sent_bytes = sockets[1].send(response)
+      self.assertEqual(sent_bytes, len(response))
+
+    def ResponseCallback(self, response, data, unpack_exception):
+      results.gotresponse = True
+      self.assertEqual(response.response_type, client.RDM_NACK_REASON)
```

Comment:
This is NACKing because it's not manufacturer specific. If you test this instead it should ACK:
http://rdm.openlighting.org/pid/display?manufacturer=31344&pid=32769

Reply:
In some ways it wasn't bad to have a NACK, but I wanted to have one that returned a response just to make sure the request actually made sense.  Needed one that was both in the dummy plugin and in the test_pids.proto and realized DMX_PERSONALITY_DESCRIPTION is, so switched to that.

Reply:
Perfect. Yeah perhaps do some magic merge/revert or manually copy/paste so we get a NACK test too?

Reply:
Actually better yet, do a GET on DMX_PERSONALITY_DESCRIPTION 0 which the responder should NACK as out of range, or even better, do a SET on DMX_PERSONALITY 0 as you're not testing that yet!

Reply:
I didn't think/explain that very well did I. We could have a GET on DMX_PERSONALITY_DESCRIPTION 0 which NACKs and a SET on DMX_PERSONALITY 1 which ACKs to test set too!

Reply:
I think at this point I'm feeling like code coverage is good enough with what's here.  If I were to do more, I think I would run the test suite with coverage and figure out what else could be tested that way.

Reply:
Fair enough.